sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -1636,11 +1637,12 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
      * snapshot, as long as the snapshot specified by previousSnapshot is the older of the two.
      * @return {@link ShareFileRange ranges} in the files that satisfy the requirements
      */
-    public PagedFlux<ShareFileRange> listRangesDiff(String previousSnapshot) {
+    public Mono<ShareFileRangeList> listRangesDiff(String previousSnapshot) {
```

Comment:
From a functional stand-point this change is fine. Generally, we have erred on using `PagedFlux`/`PagedIterable` on list responses so if the service changed in the future to be pagable we wouldn't require an API change in the SDK.

https://azure.github.io/azure-sdk/java_design.html#java-pagination-pagedflux

Reply:
We have a Mono<> for PageBlobClient.GetPageRange() and .GetPageRangeDiff(), and this API isn't currently Pagable, I think it's important that we be internally consistent..

Reply:
I believe the chance this api going pageable is small.

Reply:
Discussed offline. This is a list response but there are 2 different lists returned by the service - ClearRange and Range, so going with a Mono makes sense here.

Reply:
Didn't realize the response type was two disparate lists, given that `Mono` makes more sense.